src/renderer/utils/storageHelper.ts

```diff
@@ -0,0 +1,170 @@
+// Helper for work with persistent local storage (default: window.localStorage)
+// TODO: write unit/integration tests
+
+import type { CreateObservableOptions } from "mobx/lib/api/observable";
+import { action, comparer, observable, toJS, when } from "mobx";
+import produce, { Draft, setAutoFreeze } from "immer";
+import { isEmpty, isEqual, isFunction } from "lodash";
+
+setAutoFreeze(false); // allow to merge observables
+
+export interface StorageHelperOptions<T = any> extends StorageConfiguration<T> {
+  autoInit?: boolean; // default: true
+}
+
+export interface StorageConfiguration<T = any> {
+  storage?: StorageAdapter<T>;
+  observable?: CreateObservableOptions;
+}
+
+export interface StorageAdapter<T = any, C = StorageHelper<T>> {
+  getItem(this: C, key: string): T | Promise<T>; // import
+  setItem(this: C, key: string, value: T): void; // export
+  onChange?(this: C, value: T, oldValue?: T): void;
```

Comment:
Why does the `StorageAdaptor` need a reference to the `StorageHelper`? That seems to be an abstraction leak.

Reply:
For usability and subtle fine-tuning when needed. Don't see  any obstacles, and again this is only internal api.

Reply:
Just because it is internal doesn't mean we should have this sort of tight coupling. Can you give a good example where this sort of coupling is beneficial?

Reply:
e.g. to get access to `this.initialized` or other public internals of `StorageHelper`

Reply:
Ok, removed `this` context to for less ambiguity.